clients/src/main/java/org/apache/kafka/common/record/ByteBufferLogInputStream.java

```diff
@@ -62,13 +54,31 @@ public MutableRecordBatch nextBatch() throws IOException {
         batchSlice.limit(batchSize);
         buffer.position(buffer.position() + batchSize);
 
-        if (magic < 0 || magic > RecordBatch.CURRENT_MAGIC_VALUE)
-            throw new CorruptRecordException("Invalid magic found in record: " + magic);
-
         if (magic > RecordBatch.MAGIC_VALUE_V1)
             return new DefaultRecordBatch(batchSlice);
         else
             return new AbstractLegacyRecordBatch.ByteBufferLegacyRecordBatch(batchSlice);
     }
 
+    int nextBatchSize() throws CorruptRecordException {
+        int remaining = buffer.remaining();
+        if (remaining < LOG_OVERHEAD)
+            throw new IllegalStateException("Buffer does not contain batch header");
+
+        int recordSize = buffer.getInt(buffer.position() + SIZE_OFFSET);
+        // V0 has the smallest overhead, stricter checking is done later
+        if (recordSize < LegacyRecord.RECORD_OVERHEAD_V0)
+            throw new CorruptRecordException(String.format("Record size %d is less than the minimum record overhead (%d)",
+                    recordSize, LegacyRecord.RECORD_OVERHEAD_V0));
+        if (recordSize > maxMessageSize)
```

Comment:
What if the max message size is reduced? Is this check too strict?

Reply:
See seem to only create `ByteBufferLogInputStream` with `maxMessageSize` of `Integer.MAX_VALUE`.

Reply:
But this is a general class, right. It may be used in different ways in the future. We should try to right robust code that makes sense conceptually. If there are some assumptions, we should document them and ideally have tests so that we can't break them.